HEInventions.Logging/RavenLog.cs

```diff
@@ -0,0 +1,219 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using SharpRaven;
+using SharpRaven.Data;
+using System.Collections.Specialized;
+using Newtonsoft.Json.Linq;
+using System.IO;
+
+namespace HEInventions.Logging
+{
+    /// <summary>
+    /// This class is a wrapper for the SharpRaven module simplifying the configuration process and the 
+    /// way in which Sentry events are sent up. The class follows a Singleton pattern allowing only one instance
+    /// to be run at a time.
+    /// </summary>
+    public sealed class RavenLog
+    {
+        #region Constants
+        /// <summary>
+        /// If we cannot retrieve the machine settings, set a default value.
+        /// </summary>
+        public const string UndefinedSetting = "UNDEFINED";
+        #endregion
+
+        #region Properties and Fields
+        /// <summary>
+        /// Raven client for Sentry logging.
+        /// </summary>
+        private RavenClient ravenClient;
```

Comment:
Should this be `camelCase` because it's private?  I'm not sure about the C# convention.

Reply:
changed to _RavenClient